src/tests/java/picard/sam/markduplicates/util/OpticalDuplicateFinderTest.java

```diff
@@ -73,4 +93,26 @@ public void testParseReadName(final String readName, final int tile, final int x
                 {"RUNID:7:1203:2884:16834", 1203, 2884, 16834}
         };
     }
+
+    @Test
+    public void testVeryLongReadNames() {
```

Comment:
Could you also test that this works with the regexp version, i.e. a non-interned copy of DEFAULT_READ_NAME_REGEX?